### PR TITLE
Admin Dashboard implementing

### PR DESCRIPTION
**Pull request by it21055294**

### CODE REVIEW

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Box, Button, Grid, Paper, Typography } from '@mui/material';
+import { ComposableMap, Geographies, Geography } from 'react-simple-maps';
+
```
Comment (Mr-RiVi) [inline src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:4]: Please add the following URL to the endpoint configuration file

**Review by Mr-RiVi — COMMENTED** [review]
Overall look good. Please address the review comments and make the necessary updates. Thanks! :1st_place_medal:

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const geoUrl =
```
Comment (Mr-RiVi) [inline src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:5]: Please add the following URL to the endpoint configuration file

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                  Stress
+                </Button>
+                <Button fullWidth variant="contained" color="primary">
+                  Disease
+                </Button>
+              </Paper>
+            </Grid>
+            {/* Weather Information */}
+            <Grid item xs={12}>
+              <Paper elevation={3} sx={{ p: 2 }}>
```
[the remark this answers is not in the document]
Comment (it21055294) [reply src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:31]: going for data fetched from weather API.

File: src/compoenents/CustomerMailDetailPannel/Statistics.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  totalProjectsChange: 1.3,
+  updatedProjects: 89000,
+  updatedProjectsChange: -4.3,
+  totalPending: 2040,
+  totalPendingChange: 1.8,
+};
+
+const Statistics: React.FC = () => {
+  const [data, setData] = useState<StatsData | null>(null);
+
```
[the remark this answers is not in the document]
Comment (it21055294) [reply src/compoenents/CustomerMailDetailPannel/Statistics.tsx:61]: Yes, then go with that.

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+              </Paper>
+            </Grid>
+          </Grid>
+        </Grid>
+
+        {/* Map */}
+        <Grid item xs={12} md={9}>
+          <Paper elevation={3} sx={{ p: 2, height: '100%' }}>
+            <ComposableMap>
+              <Geographies geography={geoUrl}>
```
[the remark this answers is not in the document]
Comment (it21055294) [reply src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:46]: need more explanation

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx (hunk 2 above)
Comment (it21055294) [reply src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:5]: What URL?

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Box, Button, Grid, Paper, Typography } from '@mui/material';
+import { ComposableMap, Geographies, Geography } from 'react-simple-maps';
+
+const geoUrl =
+  "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";
```
Comment (Mr-RiVi) [inline src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:6]: Please add the following URL to the endpoint configuration file

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+          </Grid>
+        </Grid>
+
+        {/* Map */}
+        <Grid item xs={12} md={9}>
+          <Paper elevation={3} sx={{ p: 2, height: '100%' }}>
+            <ComposableMap>
+              <Geographies geography={geoUrl}>
+                {({ geographies }) =>
+                  geographies.map((geo) => (
```
Comment (Mr-RiVi) [inline src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:48]: line number 47 u used geographies parameter and line number 48 u have passed geo parameter to the arrow func cant we explicitly define the type for the geo parameter and geographies parameter? i got some type errors on those parameters. didnt u?
```suggestion
                {({ geographies: ExpectedType1 }) =>
                  geographies.map((geo: ExpectedType2) => (
```

**Review by Mr-RiVi — CHANGES_REQUESTED** [review]
Screenshot from 2024-08-07 17-16-10

Please adjust the dashboard according to the specified layout.

File: src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx (hunk 6 above)
Comment (Mr-RiVi) [inline src/compoenents/CustomerMailDetailPannel/MapDashboard.tsx:6]: Please add the following URL to the endpoint configuration file:

**Pull request closed without merging**